src/SelfService/Infrastructure/Api/RequestParser.cs

```diff
@@ -0,0 +1,141 @@
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using SelfService.Domain.Models;
+
+namespace SelfService.Infrastructure.Api;
+
+public class MissingRequestParserException<TInput> : Exception
+{
+    public MissingRequestParserException(Type outputType)
+        : base($"Unable to find a request parser for {typeof(TInput).Name} to {outputType.Name}") { }
+}
+
+public class RequestParserHelper<TInput, TOutput>
+    where TOutput : ValueObject
+{
+    private Dictionary<Type, RequestParser<TInput, TOutput>> Parsers { get; } = new();
+    private ModelStateDictionary ModalState { get; set; } = new();
+
+    private RequestParser<TInput, TOutput> GetParser(Type type)
+    {
+        if (Parsers.TryGetValue(type, out var parser))
+            return parser;
+
+        throw new MissingRequestParserException<TInput>(type);
+    }
+
+    public void Add<O>(Func<TInput, TOutput> converter)
+        where O : TOutput
+    {
+        Parsers.Add(typeof(O), new RequestParser<TInput, TOutput>(converter, typeof(O).Name));
+    }
+
+    private TOutput ParseInternal<U>(TInput i)
+        where U : TOutput
+    {
+        return GetParser(typeof(U)).TryParse(i, ModalState);
+    }
+
+    public O1 Parse<O1>(TInput i1)
+        where O1 : TOutput
+    {
+        return (O1)ParseInternal<O1>(i1);
+    }
+
+    public (O1, O2) Parse<O1, O2>(TInput i1, TInput i2)
```

Comment:
I assume this pattern of one function per parameter-count is what you did not enjoy too much yourself?

It is unfortunate, but luckily it is hidden away and not something we have to look at too much.

Reply:
Maybe one could avoid this pattern by using some variation of the builder pattern.
PseudoCode
```
var (_, _) = RequestParserRegistry
            .StringToValueParser(dictionary)
            .Parse<CapabilityId>(validCapabilityId)
            .Parse<KafkaTopicId>(validKafkaTopicId)
            .Parse<KafkaTopicId>(otherValidTopic)
            .ExecuteParse()
```

It is still easier to read than the current version, but it may be simpler to maintain and understand?
I even think we could get rid of the ExecuteParse function.